src/utils/jump_utils.jl

```diff
@@ -44,57 +45,162 @@ function fix_parameter_value(input::JuMP.VariableRef, value::Float64)
+  a number of rows equal to the length of the input and one column.
+- If the input is a 2d DenseAxisArray, the dimensions are transposed, due to the way we
+  store outputs in JuMP.
+"""
+function to_matrix(vec::Vector)
+    data = vec[:]
+    return reshape(data, length(data), 1)
+end
+
+to_matrix(array::Matrix) = array
+
+to_matrix(array::Matrix) = array
+
+function to_matrix(array::DenseAxisArray{T, 1}) where {T}
     data = array.data[:]
     return reshape(data, length(data), 1)
 end
 
-to_matrix(array::DenseAxisArray{T, 2} where {T}) = permutedims(array.data)
+function to_matrix(array::DenseAxisArray{T, 2}) where {T}
+    return permutedims(array.data)
+end
 
-function to_matrix(::DenseAxisArray{T, N, K}) where {T, N, K <: NTuple{N, Any}}
-    error(
-        "Converting $(N)-dimensional DenseAxisArrays to matrix is currently not supported",
-    )
+function to_matrix(array::DenseAxisArray)
+    error("Converting type = $(typeof(array)) to a matrix is not supported.")
 end
 
-function get_column_names(key::OptimizationContainerKey)
-    return ([encode_key_as_string(key)],)
+function _to_matrix(
+    array::SparseAxisArray{T, N, K},
+    columns,
+) where {T, N, K <: NTuple{N, Any}}
+    time_steps = Set{Int}(k[N] for k in keys(array.data))
+    data = Matrix{Float64}(undef, length(time_steps), length(columns))
+    for (ix, col) in enumerate(columns), t in time_steps
+        data[t, ix] = array.data[(col..., t)]
+    end
+    return data
 end
 
-function get_column_names(
-    key::OptimizationContainerKey,
-    ::DenseAxisArray{T, 1, K},
-) where {T, K <: NTuple{1, Any}}
-    return get_column_names(key)
+function to_matrix(array::SparseAxisArray{T, N, K}) where {T, N, K <: NTuple{N, Any}}
+    # Don't use get_column_names_from_axis_array to avoid additional string conversion
+    # TODO DT: I don't understand why we have two mechanisms of creating columns.
+    # Why does get_column_names_from_axis_array rely on encode_tuple_to_column?
+    columns = sort!(unique!([k[1:(N - 1)] for k in keys(array.data)]))
+    return _to_matrix(array, columns)
 end
 
-function get_column_names(
-    k::OptimizationContainerKey,
-    array::DenseAxisArray{T, 2, K},
-) where {T, K <: NTuple{2, Any}}
-    return (string.(axes(array)[1]),)
+"""
+Return column names from the axes of a JuMP DenseAxisArray or SparseAxisArray.
+The columns are returned as a tuple of vector of strings.
+1d and 2d arrays will return a tuple of length 1.
+3d arrays will return a tuple of length 2.
+
+There are two variants of this function:
+  - get_column_names_from_axis_array(array)
+  - get_column_names_from_axis_array(::OptimizationContainerKey, array)
+  
+When the variant with the key is called:
+  In cases where the array has one dimension, retrieve the column names from the key.
+  In cases where the array has two or more dimensions, retrieve the column names from the
+  axes.
+"""
+# TODO DT: the docstring describes what the code does.
```

Comment:
After going through all this, I think that we should change it so that we don't need the key. We can always store the column names in the axes. I tried to maintain current behavior in order not to break things.